src/components/Gists.js

```diff
@@ -1,105 +1,89 @@
 import React, { Component } from 'react';
+import _ from 'lodash';
 import Container from '@material-ui/core/Container';
 import Grid from '@material-ui/core/Grid';
 import Gist from './Gist';
-import { TEXT, PYTHON, XML } from '../config/constants/FileTypes';
+import fileMap from '../constants/fileMap';
+import { getUserGistsUrl } from '../config/api';
 
 class Gists extends Component {
   state = {
-    message: 'Loading gists',
-    gists: {}
+    emptyMessage: 'Loading gists',
+    gists: []
   };
   
   componentDidMount() {
     // Load gists
-    let userId = this.props.match.params.user_id;
-    let url = 'https://api.github.com/users/' + userId + '/gists?page=0&per_page=10';
-
-    fetch(url).then((res) => {
-      // Check status
-      if (res.status >= 200 && res.status < 300) {
-        return res.json();
-      }
-      this.setState({ message: 'No gists found' });
-      return Promise.reject();
-    }).then((data) => {
-      // Load gist data
-      let gists = {};
-
-      // Create list of gists
-      if (data.length > 0) {
-        data.forEach(gist => {
-          let files = [];
-          // Create list of files for each gist
-          Object.values(gist.files).forEach(file => {
-            let type = 'Unknown';
+    this.fetchGistData();
+  }
 
-            // Parse file type
-            switch(file.type) {
-              case TEXT:
-                type = "Text";
-                break;
-              case PYTHON:
-                type = "Python";
-                break;
-              case XML:
-                type = "XML";
-                break;
-            }
+  fetchGistData() {
+    let userId = _.get(this.props, 'match.params.user_id');
+    let url = getUserGistsUrl(userId, 0, 10);
+    
+    fetch(url)
+      .then(res => {
+        // Check status
+        if (res.status >= 200 && res.status < 300) {
+          return res.json();
+        }
+        this.setState({ emptyMessage: 'Invalid username' });
+        return Promise.reject();
+      })
+      .then(data => {
+        if (data === undefined) {
+          console.log('Returned data is undefined');
+          return Promise.reject();
+        }
 
-            files.push({
-              filename: file.filename,
-              url: file.raw_url,
-              type: type
-            });
+        // Create a parsed list of gists
+        let gists = _.map(data, gist => {
+          // Create a new list of truncated file data for each gist
+          let files = _.map(gist.files, file => {
+            let type = _.get(fileMap, file.type, 'Unknown'); // Parsing file type
+            return { filename: file.filename, url: file.raw_url, type}; // Returning updated file object
           });
 
-          gists[gist.id] = {
-            ...gist,
-            username: gist.owner.login,
```

Comment:
this strategy in Python is called Monkey Patching. you can read further by reading the documentation of SQLAlchemy. It's also published in chapter 21 of the Architecture of Open Source Application.